tests/test_utils.py

```diff
@@ -0,0 +1,132 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+"""Test cases for annotations."""
+
+import unittest
+import copy
+import itertools
+import numpy as np
+
+import astropy.units as u
+import astropy.time
+from astro_metadata_translator import makeObservationInfo
+
+import lsst.afw.image as afwImage
+import lsst.geom as geom
+from lsst.obs.base import createInitialSkyWcsFromBoresight
+from lsst.obs.base.makeRawVisitInfoViaObsInfo import MakeRawVisitInfoViaObsInfo
+from lsst.obs.lsst.latiss import Latiss
+import lsst.utils.tests
+from lsst.rapid.analysis.utils import getExpPositionOffset
+
+
+class ExpSkyPositionOffsetTestCase(lsst.utils.tests.TestCase):
+    """A test case for testing sky position offsets for exposures."""
+
+    def setUp(self):
+        camera = Latiss.getCamera()
+        self.assertTrue(len(camera) == 1)
+        self.detector = camera[0]
+
+        self.viMaker = MakeRawVisitInfoViaObsInfo()
+        self.mi = afwImage.maskedImage.MaskedImageF(0, 0)
+        self.baseHeader = dict(boresight_airmass=1.5,
+                               temperature=15*u.deg_C,
+                               observation_type="science",
+                               exposure_time=5*u.ks,
+                               detector_num=32,
+                               )
+
+    def test_getExpPositionOffset(self):
+        epsilon = 0.0001
+        ra1s = [0, 45, 90]
+        ra2s = copy.copy(ra1s)
+        ra2s.extend([r + epsilon for r in ra1s])
+        ra1s = np.deg2rad(ra1s)
+        ra2s = np.deg2rad(ra2s)
+
+        epsilon = 0.0001
+        dec1s = [0, 45, 90]
+        dec2s = copy.copy(dec1s)
+        dec2s.extend([d + epsilon for d in dec1s[:-1]])  # skip last point as >90 not allowed for dec
+
+        rotAngle1 = geom.Angle(43.2, geom.degrees)  # arbitrary non-zero
+        rotAngle2 = geom.Angle(56.7, geom.degrees)
+
+        t1 = astropy.time.Time("2021-09-15T12:00:00", format="isot", scale="utc")
+        t2 = astropy.time.Time("2021-09-15T12:01:00", format="isot", scale="utc")
+        expTime = astropy.time.TimeDelta(20, format='sec')
+
+        header1 = copy.copy(self.baseHeader)
+        header2 = copy.copy(self.baseHeader)
+        header1['datetime_begin'] = astropy.time.Time(t1, format="isot", scale="utc")
+        header2['datetime_begin'] = astropy.time.Time(t2, format="isot", scale="utc")
+
+        header1['datetime_end'] = astropy.time.Time(t1+expTime, format="isot", scale="utc")
+        header2['datetime_end'] = astropy.time.Time(t2+expTime, format="isot", scale="utc")
+
+        obsInfo1 = makeObservationInfo(**header1)
+        obsInfo2 = makeObservationInfo(**header2)
```

Comment:
It's a shame you can't use a real header, then generate an ObservationInfo that has reference AltAz and SkyCoord objects in it. Since in the test you calculate a WCS from some known RA/Decs if you did that from ObservationInfo you would be able to test that your code is consistent with ObservationInfo and also test the deltaAz/El calculation.